xarray/indexes/range_index.py

```diff
@@ -0,0 +1,352 @@
+
+        >>> index = RangeIndex.arange("x", "x", 0.0, 1.0, 0.2)
+        >>> ds = xr.Dataset(coords=xr.Coordinates.from_xindex(index))
+
+        >>> ds
+        <xarray.Dataset> Size: 40B
+        Dimensions:  (x: 5)
+        Coordinates:
+          * x        (x) float64 40B 0.0 0.2 0.4 0.6 0.8
+        Data variables:
+            *empty*
+        Indexes:
+            x        RangeIndex (start=0, stop=1, step=0.2)
+
+        """
+        size = math.ceil((stop - start) / step)
+
+        transform = RangeCoordinateTransform(
+            start, stop, size, coord_name, dim, dtype=dtype
+        )
+
+        return cls(transform)
+
+    @classmethod
+    def linspace(
+        cls,
+        coord_name: Hashable,
+        dim: str,
+        start: float,
+        stop: float,
+        num: int = 50,
+        endpoint: bool = True,
+        dtype: Any = None,
+    ) -> "RangeIndex":
+        """Create a new RangeIndex from given start / stop values and number of
+        values.
+
+        Parameters
+        ----------
+        coord_name : Hashable
+            Name of the (lazy) coordinate variable that will be created and
+            associated with the new index.
+        dim : str
+            Dimension name.
+        start : float
+            Start of interval. The interval includes this value.
+        stop : float, optional
+            End of interval. The interval includes this value if ``endpoint=True``.
+        num : float, optional
+            Number of values in the interval, i.e., dimension size (default: 50).
+        endpoint : bool, optional
+            If True (default), the ``stop`` value is included in the interval.
+        dtype : dtype, optional
+            The dtype of the coordinate variable (default: float64).
+
+        Examples
+        --------
+        >>> from xarray.indexes import RangeIndex
+
+        >>> index = RangeIndex.linspace("x", "x", 0.0, 1.0, 5)
+        >>> ds = xr.Dataset(coords=xr.Coordinates.from_xindex(index))
+
+        >>> ds
+        <xarray.Dataset> Size: 40B
+        Dimensions:  (x: 5)
+        Coordinates:
+          * x        (x) float64 40B 0.0 0.25 0.5 0.75 1.0
+        Data variables:
+            *empty*
+        Indexes:
+            x        RangeIndex (start=0, stop=1.25, step=0.25)
+
+        """
+        if endpoint:
+            stop += (stop - start) / (num - 1)
+
+        transform = RangeCoordinateTransform(
+            start, stop, num, coord_name, dim, dtype=dtype
+        )
+
+        return cls(transform)
+
+    @classmethod
+    def from_variables(
+        cls,
+        variables: Mapping[Any, Variable],
+        *,
+        options: Mapping[str, Any],
+    ) -> "RangeIndex":
+        raise NotImplementedError(
+            "cannot create a new RangeIndex from an existing coordinate. Use instead "
+            "either `RangeIndex.arange()` or `RangeIndex.linspace()` together with "
+            "`Coordinates.from_xindex()`"
+        )
+
+    @property
```

Comment:
Can these all be `cached_property`?

Reply:
Would there be much benefit of caching those simple aliases to attributes of the underlying `transform`?